<launch>
    <arg name="vehicle"             default="H01"/>
    <arg name="enable_beacon"       default="1"/>
    <arg name="num_beacons_brd_0"   default="3"/>
    <arg name="num_beacons_brd_1"   default="3"/>

    <group ns="$(arg vehicle)" if="$(arg enable_beacon)">
        <!-- ROSSERIAL PYTHON -->
        <node pkg="rosserial_python" type="serial_node.py" name="beacon_dropper_0" output="log">
            <param name="port" value="/dev/teensy_b0_first_port"/>
        </node>
    
        <!-- ROSSERIAL PYTHON -->
        <node pkg="rosserial_python" type="serial_node.py" name="beacon_dropper_1" output="log">
            <param name="port" value="/dev/teensy_b1_first_port"/>
        </node>
    </group>

    <node pkg="beacon_combiner" type = "beacon_brd_combiner.py" name="combiner">
        <param name="vehicle_name" value = "$(arg vehicle)"/>
        <param name="num_beacons_brd_0" value = "$(arg num_beacons_brd_0)"/>
        <param name="num_beacons_brd_1" value = "$(arg num_beacons_brd_1)"/>
    </node>

</launch>
